Library/SPIClient.m

```diff
@@ -1079,11 +1089,35 @@ - (void)autoResolveEftposAddress {
     }
     
     [[SPIDeviceService alloc] retrieveServiceWithSerialNumber:_serialNumber apiKey:_deviceApiKey acquirerCode:_acquirerCode isTestMode:_testMode completion:^(SPIDeviceAddressStatus *addressResponse) {
-        if (addressResponse.address.length == 0) {
+        if (addressResponse == nil) {
+            SPIDeviceAddressStatus *currentDeviceAddressStatus = [SPIDeviceAddressStatus new];
+            currentDeviceAddressStatus.deviceAddressResponseCode = DeviceAddressResponceCodeDeviceError;
+            self.state.deviceAddressStatus = currentDeviceAddressStatus;
+            [self deviceAddressChanged];
             return;
         }
         
+        if (addressResponse.address.length == 0) {
+            if (addressResponse.responseCode == 404) {
+                SPIDeviceAddressStatus *currentDeviceAddressStatus = [SPIDeviceAddressStatus new];
```

Comment:
The only thing different between these two branches here is the `deviceAddressResponseCode` line. Extract the instantiation of `currentDeviceAddressStatus` to before the if statement and the rest after.

Reply:
Fixed